index.js

```diff
@@ -62,6 +81,10 @@ if (!Cache.prototype.addAll) {
       // (don't think this is possible to polyfill due to opaque responses)
       return Promise.all(
         responses.map(function(response, i) {
+          if (!response.ok) {
+            throw new NetworkError('Incorrect response status');
+          }
```

Comment:
None of the responses should cache if any are `!ok`, could you check all of them before calling `.put`?